x/bridge/observer/bitcoin/bitcoin.go

```diff
@@ -0,0 +1,301 @@
+		}
+		return errorsmod.Wrapf(err, "Failed to get block hash")
+	}
+
+	b.logger.With("height", nextHeight, "hash", hash).Info("Fetching bitcoin block")
+
+	blockVerbose, err := b.btcRpc.GetBlockVerboseTx(hash)
+	if err != nil {
+		return errorsmod.Wrapf(err, "Failed to get verbose block")
+	}
+
+	for _, tx := range blockVerbose.Tx {
+		txIn, isRelevant, err := b.processTx(uint64(blockVerbose.Height), &tx)
+		if err != nil {
+			b.logger.Error(fmt.Sprintf("Failed to process Tx %s: %s", tx.Txid, err.Error()))
+			continue
+		}
+		if !isRelevant {
+			continue
+		}
+
+		select {
+		case b.outboundChan <- txIn:
+		case <-b.stopChan:
+			b.logger.Info("Observer exiting early, tx skipped: ", tx.Txid)
+			return nil
+		}
+	}
+
+	b.lastObservedHeight.Add(1)
+	return nil
+}
+
+// processTx builds `Transfer` from provided Bitcoin transaction
+func (b *ChainClient) processTx(height uint64, tx *btcjson.TxRawResult) (observer.Transfer, bool, error) {
+	amount, relevant := b.isTxRelevant(tx)
+	if !relevant {
+		return observer.Transfer{}, false, nil
+	}
+
+	memo, contains := getMemo(tx)
+	if !contains {
+		return observer.Transfer{}, false, fmt.Errorf("failed to get Tx memo")
+	}
+
+	return observer.Transfer{
+		SrcChain: observer.ChainIdBitcoin,
+		DstChain: observer.ChainIdOsmosis,
+		Id:       tx.Hash,
+		Height:   height,
+		Sender:   "", // NB! the sender is set in the osmosis chain client
+		To:       memo,
+		Asset:    bridgetypes.DefaultBitcoinDenomName,
+		Amount:   amount,
+	}, true, nil
+}
+
+// isTxRelevant checks if a tx contains transfers to the BTC vault and calculates the total amount to transfer.
+// This method goes through all the Vouts and accumulates their values if Vouts.account[0] == vault.
+// Returns true is the tx is not relevant. Otherwise, false and the total amount to transfer.
+func (b *ChainClient) isTxRelevant(tx *btcjson.TxRawResult) (math.Uint, bool) {
+	var amount = sdk.NewUint(0)
+	for _, vout := range tx.Vout {
+		if strings.EqualFold(vout.ScriptPubKey.Type, "nulldata") {
+			continue
+		}
+		if vout.Value <= 0 {
+			continue
+		}
+
+		addresses, err := b.getAddressesFromScriptPubKey(vout.ScriptPubKey)
+		if err != nil || len(addresses) != 1 {
+			continue
+		}
+
+		if slices.Contains(addresses, b.vaultAddr) {
+			a, err := getAmount(vout)
+			if err != nil {
+				continue
+			}
+			amount = amount.Add(a)
+		}
+	}
+	return amount, !amount.IsZero()
+}
+
+// getAmount retrieves amount of tokens sent
+func getAmount(vout btcjson.Vout) (math.Uint, error) {
+	amount, err := btcutil.NewAmount(vout.Value)
+	if err != nil {
+		return math.Uint{}, errorsmod.Wrapf(err, "Failed to parse float value")
+	}
+	return math.NewUint(uint64(amount.ToUnit(btcutil.AmountSatoshi))), nil
+}
+
+// getMemo retrieves data behind `OP_RETURN` Vout. Return false if can't get memo.
+func getMemo(tx *btcjson.TxRawResult) (string, bool) {
+	for _, vout := range tx.Vout {
+		if !strings.EqualFold(vout.ScriptPubKey.Type, "nulldata") {
+			continue
+		}
```

Comment:
Can we explain why we are skipping the vout if the ScriptPubKey type is not equal to nulldata?